data-access/mandataris-download.ts

```diff
@@ -0,0 +1,257 @@
+  } = filters;
+  let bestuursorgaanInTijdFilter: string | null = null;
+  let onlyActiveFilter: string | null = null;
+  let persoonFilter: string | null = null;
+  let mandaatTypeFilter: string | null = null;
+
+  if (bestuursorgaanId) {
+    bestuursorgaanInTijdFilter = `
+      ?bestuursorgaanInTijd org:hasPost ?mandaat.
+      ?bestuursorgaanInTijd mu:uuid ${sparqlEscapeString(bestuursorgaanId)}.
+    `;
+  }
+  if (filters.activeOnly) {
+    const escapedTodaysDate = sparqlEscapeDateTime(new Date());
+    onlyActiveFilter = `
+      ?mandataris mandaat:einde ?einde.
+
+      ?bestuursorgaanInTijd mandaat:bindingStart ?startBestuursorgaan. 
+      ?bestuursorgaanInTijd mandaat:bindingEinde ?eindeBestuursorgaan. 
+
+      FILTER (
+        ?einde <= ?eindeBestuursorgaan &&
+        ?einde >= ?startBestuursorgaan &&
+        ?einde >= ${escapedTodaysDate}
+      )
+      BIND(IF(BOUND(?eindeBestuursorgaan), ?eindeBestuursorgaan, ${escapedTodaysDate}) AS ?saveEindeBestuursorgaan).
+    `;
+  }
+
+  if (persoonIds.length >= 1) {
+    const idValues = persoonIds.map((id) => sparqlEscapeString(id)).join('\n');
+    persoonFilter = `
+      VALUES ?persoonId { ${idValues} }
+
+      ?mandataris mandaat:isBestuurlijkeAliasVan ?persoon.
+      ?persoon mu:uuid ?persoonId.
+    `;
+  }
+
+  if (bestuursFunctieCodeIds.length >= 1) {
+    const idValues = bestuursFunctieCodeIds
+      .map((id) => sparqlEscapeString(id))
+      .join('\n');
+    mandaatTypeFilter = `
+      VALUES ?functieCode { ${idValues} }
+      ?mandaat org:role ?bestuursfunctieCode .
+      ?bestuursfunctieCode mu:uuid ?functieCode.
+    `;
+  }
+
+  const queryString = `
+    PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+    PREFIX lmb: <http://lblod.data.gift/vocabularies/lmb/>
+    PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+    PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+    PREFIX fractieType: <http://data.vlaanderen.be/id/concept/Fractietype/>
+
+
+    SELECT DISTINCT ?mandataris
+    WHERE {
+      ?mandataris a mandaat:Mandataris.
+      ?mandataris org:holds ?mandaat.
+
+      ?bestuursorgaan lmb:heeftBestuursperiode ?bestuursperiode.
+      ?bestuursperiode mu:uuid ${sparqlEscapeString(bestuursperiodeId)}.
+
+      ${bestuursorgaanInTijdFilter ?? ''}
+      ${onlyActiveFilter ?? ''}
+      ${persoonFilter ?? ''}
+      ${getFractieFilters(filters)}
+      ${mandaatTypeFilter ?? ''}
+    }
+  `;
+
+  const sparqlResult = await query(queryString);
+
+  return getSparqlResults(sparqlResult).map((res) => res.mandataris.value);
+}
+
+function getFractieFilters(filters): string {
+  const {
+    fractieIds,
+    hasFilterOnOnafhankelijkeFractie,
+    hasFilterOnNietBeschikbareFractie,
+  } = filters;
+  const idValues = fractieIds.map((id) => sparqlEscapeString(id)).join('\n');
+  const unions = [
+    {
+      apply:
+        fractieIds.length === 0 &&
```

Comment:
i think if this condition is true, then you can return the empty string as a fraction filter? sounds like an early return

Reply:
Done